bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/controller/ChangelogController.kt

```diff
@@ -0,0 +1,87 @@
+import org.eclipse.tractusx.bpdm.common.dto.request.PaginationRequest
+import org.eclipse.tractusx.bpdm.common.dto.response.PageResponse
+import org.eclipse.tractusx.bpdm.gate.dto.response.LsaType
+import org.eclipse.tractusx.bpdm.gate.dto.response.PageChangeLogResponse
+import org.eclipse.tractusx.bpdm.gate.entity.ChangelogEntity
+import org.eclipse.tractusx.bpdm.gate.service.ChangelogService
+import org.springdoc.core.annotations.ParameterObject
+import org.springframework.validation.annotation.Validated
+import org.springframework.web.bind.annotation.*
+import java.time.Instant
+
+@RestController
+@RequestMapping("/api/catena/business-partners/changelog")
+@Validated
+class ChangelogController (
+    private val  changelogService: ChangelogService
+    ) {
+
+    @Operation(
+        summary = "Get business partner changelog entries by list external id, from timestamp",
+        description = "Get business partner changelog entries by list external id, from timestamp"
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "The changelog entries for the specified parameters"),
+            ApiResponse(responseCode = "400", description = "On malformed pagination request", content = [Content()]),
+            ApiResponse(responseCode = "404", description = "No business partner found for specified bpn", content = [Content()])
+        ]
+    )
+    @PostMapping("/search")
+    fun getChangelogEntriesExternalId(
+        @ParameterObject @Valid paginationRequest: PaginationRequest,
+        @Parameter(description = "From Time", example = "2023-03-20T10:23:28.194Z") @RequestParam(required = false) fromTime: Instant?,
+        @RequestBody(required = true)  @NotEmpty(message = "Input externalIds list cannot be empty.") externalIds: Collection<String>
+    ): PageChangeLogResponse<ChangelogEntity> {
+        return changelogService.getChangeLogByExternalId(externalIds,fromTime,paginationRequest.page,paginationRequest.size)
+    }
+    @Operation(
+        summary = "Get business partner changelog entries by timestamp or LSA type",
+        description = "Get business partner changelog entries by from timestamp or LSA type"
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "The changelog entries for the specified parameters"),
+            ApiResponse(responseCode = "400", description = "On malformed pagination request", content = [Content()]),
+            ApiResponse(responseCode = "404", description = "No business partner found for specified bpn", content = [Content()])
+        ]
+    )
+    @PostMapping("/filter")
+    fun getChangelogEntriesLsaType(
+        @ParameterObject @Valid paginationRequest: PaginationRequest,
+        @Parameter(description = "From Time", example = "2023-03-20T10:23:28.194Z") @RequestParam fromTime: Instant?,
+        @Parameter(description = "LSA Type") @RequestParam lsaType: LsaType?
```

Comment:
LsaType is not a good enumeration to take here as it contains "None". Would need a new enumeration just containing the actual three LSA types